pkg/clients/clients.go

```diff
@@ -375,6 +141,55 @@ func (dcs *DynamicClientSet) ResourceClientForObject(obj *unstructured.Unstructu
 	return dcs.ResourceClient(obj.GroupVersionKind(), obj.GetNamespace())
 }
 
+func (dcs *DynamicClientSet) gvkForKind(kind Kind) (gvk schema.GroupVersionKind, err error) {
+	resources, err := dcs.DiscoveryClientCached.ServerPreferredResources()
```

Comment:
Just so I'm clear about this: we're using `ServerPreferredResources` here because the API server, when used with this client, transparently converts various API versions into the "preferred" one, right? Also which versions of k8s does this work with.

Reply:
Yes, that's my understanding. I've actually modified this logic somewhat so that it uses the provided GV by default. I think it should work for any k8s.